src/vm/gcenv.ee.cpp

```diff
@@ -1551,3 +1551,45 @@ void GCToEEInterface::VerifySyncTableEntry()
     SyncBlockCache::GetSyncBlockCache()->VerifySyncTableEntry();
 #endif // VERIFY_HEAP
 }
+
+#if defined(__linux__)
+void GCToEEInterface::UpdateGCEventStatus()
```

Comment:
>UpdateGCEventStatus [](start = 22, length = 19)

I would keep a record of the provider/keyword/level and only when that changes would I actually call GCHeapUtilities::RecordEventStateChange.